src/test/groovy/de/dkfz/roddy/execution/jobs/BatchEuphoriaJobManagerSpec.groovy

```diff
@@ -0,0 +1,174 @@
+package de.dkfz.roddy.execution.jobs
+
+import de.dkfz.roddy.TestExecutionService
+import de.dkfz.roddy.execution.jobs.cluster.JobManagerImplementationBaseSpec
+import spock.lang.Shared
+
+import static de.dkfz.roddy.execution.jobs.JobState.*
+
+class BatchEuphoriaJobManagerSpec extends JobManagerImplementationBaseSpec {
+
+    /**
+     * We will only test non abstract methods in the BatchEuphoriaJobManager class (which is abstract for reasons)
+     * For this, we will first create a custom test class, overriding some special methods necessary for our test
+     * cases. Afterwards, we create a Spy() of this class, which can then be utilized in our tests.
+     */
+    abstract class TestBEJobManager extends BatchEuphoriaJobManager {
+        TestBEJobManager() {
+            super(
+                    new TestExecutionService("test", "test"),
+                    new JobManagerOptionsBuilder().build()
+            )
+        }
+
+        @Override
+        Map<BEJobID, JobInfo> queryJobInfo(List jobIDs) {
+            // For the tests, it actually does not matter if the result contains JobInfo or ExtendedJobInfo objects
+            return queryExtendedJobInfo(jobIDs) as Map<BEJobID, JobInfo>
+        }
+
+        @Override
+        Map<BEJobID, ExtendedJobInfo> queryExtendedJobInfo(List jobIDs) {
+            [
+                    "1000": RUNNING,
+                    "1001": SUSPENDED,
+                    "1002": COMPLETED_SUCCESSFUL,
+                    "1003": FAILED,
+            ].collectEntries {
+                String _id, JobState state ->
+                    BEJobID id = new BEJobID(_id)
+                    [id, new ExtendedJobInfo(id, state)]
+            } as Map<BEJobID, ExtendedJobInfo>
+        }
+    }
+
+    def "test assertListIsValid"(list) {
+        when:
+        BatchEuphoriaJobManager.assertListIsValid(list)
+
+        then:
+        thrown(AssertionError)
+
+        where:
+        list   | _
+        null   | _
+        []     | _
+        [null] | _
+    }
+
+    @Shared
+    TestBEJobManager jobManager = Spy()
+
+    def "test queryJobInfoByJob (query for single object, also tests query by ID!)"(String id, JobState expectedState) {
+        when:
+        JobInfo result = jobManager.queryJobInfoByJob(new BEJob(new BEJobID(id), jobManager))
+
+        then:
+        result          // There is ALWAYS a result!
+        result.jobState == expectedState
+
+        where:
+        id     | expectedState
+        "999"  | UNKNOWN
+        "1000" | RUNNING
+        "1001" | SUSPENDED
+        "1002" | COMPLETED_SUCCESSFUL
```

Comment:
Should we actually also start fixing the states? Maybe soon. The states currently code two pieces of information in the name, namely status according to batch processing system (RUNNING, UNKOWN, COMPLETED) and exit code (SUCCESSFUL, FAILURE). We should absorb the exit code into a field.